File: book-service/src/main/java/com/bookservice/exception/NotFoundException.java
package com.bookservice.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(code = HttpStatus.NOT_FOUND)
public class NotFoundException extends RuntimeException{

    private final ExceptionMessage message;
    public NotFoundException(ExceptionMessage message) {
        this.message = message;
    }

    public ExceptionMessage getExceptionMessage() {
        return message;
    }
}

File: review-service/src/main/kotlin/reviewservice/exception/NotFoundException.kt
package reviewservice.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(code = HttpStatus.NOT_FOUND, reason = "Not Found")
class NotFoundException(message: ExceptionMessage) : RuntimeException() {

    private val exceptionMessage: ExceptionMessage = message

    fun getMessage(): ExceptionMessage{
        return exceptionMessage
    }
}